spec/controllers/state_file/questions/az_qualifying_organization_contributions_controller_spec.rb

```diff
@@ -213,6 +213,9 @@
       get :index
       expect(response.body).not_to include(I18n.t('state_file.questions.az_qualifying_organization_contributions.index.maximum_records'))
 
+      # remove the built-but-invalid az321_contribution relation created when #index renders the :new template
+      subject.current_intake.reload
```

Comment:
is this not a problem in the real implementation? does the `current_intake` reload on the create action in the controller?

Reply:
I spent a while figuring out what was going on here :(

As I understand it, this problem will never pop up in real usage. The issue is that the `:index`/`:new` actions create blank contributions for the form to use when rendering. However, because we call multiple actions within a single test, the controller (and its references) are never garbage collected, and so when `create` is called, the blank (and therefore invalid) contribution from `index` is still present in the `current_intake`'s `az321_contributions` relation. This prevents the subsequent contributions from being saved, because the intake is invalid by virtue of the blank unsaved contribution in its relation. In production, new instances of the controller would be created for each request so this could not occur. The `reload` simulates a new controller instance being created for the subsequent requests, as would happen in a real deployment.

I've confirmed this understanding by calling `get :new` before making the create calls in the test code without the other changes in this PR and reproducing the failure I saw when I added the blank contribution creation to `index`.